lib/friendly_shipping/services/usps/parse_address_validation_response.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module FriendlyShipping
+  module Services
+    class Usps
+      class ParseAddressValidationResponse
+        class << self
+          # Parse a response from USPS' address validation API
+          #
+          # @param [FriendlyShipping::Request] request The request that was used to obtain this Response
+          # @param [FriendlyShipping::Response] response The response that USPS returned
+          # @param [Physical::Location] location The location object we're trying to get results for
+          # @return [Result<FriendlyShipping::AddressValidationResult>]
+          def call(request:, response:, location:)
+            # Filter out error responses and directly return a failure
+            parsing_result = ParseXMLResponse.call(response.body, 'AddressValidateResponse')
+            parsing_result.fmap do |xml|
+              address = xml.root.at('Address')
+              suggestions = [
+                Physical::Location.new(
+                  address1: address&.at('Address2')&.text, # USPS swaps Address1 and Address2 in the response
```

Comment:
This is actually not true. As it turns out, USPS simply expects Address1 to be the second line of the address, and Address2 to be the first line of the address. It wasn't just swapping them in the response. I was actually passing them incorrectly, but they just happened to validate correctly 99% of the time. I need to make this change in the CS app as well and will create a new PR for this project that makes the same adjustment.

Reply:
Actually, disregard. I see now that this is correct behavior since a Friendly Shipping address stores line 1 as line 1, and line 2 as line 2, as any person with common sense would. So we are having to swap what we send to USPS in addition to what we get back. The way the comment is worded was what confused me. It should probably say: "USPS expects address line 1 in Address2 and address line 2 in Address1"